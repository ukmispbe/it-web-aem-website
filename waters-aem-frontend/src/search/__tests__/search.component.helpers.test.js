import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import props from '../__mocks__/en_US/';
import { defaultProps } from '../search.component.props';
import { FilterTagList } from '../search.component.helpers';

describe('Feature: FilterTagList Component', () => {
    const propsWithNoFilters = {
        text: { ... props.searchText },
        filterMap: { ... props.filterMap },
        filterTagsProps: { ... defaultProps.filterTagsProps },
        filterTagsEvents: { ... defaultProps.filterTagsEvents }
    };

    const propsWithKeyword = {
        text: { ... props.searchText },
        filterMap: { ... props.filterMap },
        filterTagsProps: {
            keyword: "hplc",
            contentTypeSelected: {},
            selectedFacets: {}
        },
        filterTagsEvents: { ... defaultProps.filterTagsEvents }
    };

    const propsWithContentType = {
        text: { ... props.searchText },
        filterMap: { ... props.filterMap },
        filterTagsProps: {
            keyword: "",
            contentTypeSelected: {
                facetName: "applicationnote_facet", 
                facetValue: "Application Note", 
                facetTranslation: "Application Note"
            },
            selectedFacets: {}
        },
        filterTagsEvents: { ... defaultProps.filterTagsEvents }
    };

    const propsWithSubFacets = {
        text: { ... props.searchText },
        filterMap: [],
        filterTagsProps: {
            keyword: "",
            contentTypeSelected: {
                facetName: "applicationnote_facet", 
                facetValue: "Application Note", 
                facetTranslation: "Application Note"
            },
            selectedFacets: {
                separationmode_facet: ["Reversed-phase"]
            }
        },
        filterTagsEvents: { ... defaultProps.filterTagsEvents }
    };

    const propsWithAllFilters = {
        text: { ... props.searchText },
        filterMap: [],
        filterTagsProps: {
            keyword: "hplc",
            contentTypeSelected: {
                facetName: "applicationnote_facet", 
                facetValue: "Application Note", 
                facetTranslation: "Application Note"
            },
            selectedFacets: {
                separationmode_facet: ["Reversed-phase"]
            }
        },
        filterTagsEvents: { 
            onClearAll: jest.fn(),
            onKeywordRemove: jest.fn(),
            onContentTypeRemove: jest.fn(),
            onSubFacetRemove: jest.fn()
         }
    };

    describe('Scenario: Rendering', () => {
        describe('When no filters are specified', () => {
            let wrapper;

            beforeAll(() => {
                wrapper = shallow(<FilterTagList {...propsWithNoFilters} />);
            });

            it('Then it should match snapshot', () => {
                const json = renderer.create(<FilterTagList {...propsWithNoFilters} />);

                expect(json).toMatchSnapshot();
            });

            it('And it should not contain a keyword tag', () => {
                const tag = wrapper.find('KeywordTag');

                expect(tag.exists()).toEqual(false);
            });

            it('And it should not contain a content type tag', () => {
                const tag = wrapper.find('ContentTypeTag');

                expect(tag.exists()).toEqual(false);
            });

            it('And it should not contain sub facet tags', () => {
                const tag = wrapper.find('SubFacetTags');

                expect(tag.exists()).toEqual(false);
            });

            it('And it should not contain a clear all tag', () => {
                const tag = wrapper.find('ClearAllTag');

                expect(tag.exists()).toEqual(false);
            });
        });

        describe('When keyword filter is specified', () => {
            let wrapper;

            beforeAll(() => {
                wrapper = shallow(<FilterTagList {...propsWithKeyword} />);
            });

            it('Then it should match spapshot', () => {
                const json = renderer.create(<FilterTagList {...propsWithKeyword} />);

                expect(json).toMatchSnapshot();
            });

            it('And it should contain a keyword tag', () => {
                const tag = wrapper.find('KeywordTag');

                expect(tag.exists()).toEqual(true);
            });

            it('And it should contain a clear all tag', () => {
                const tag = wrapper.find('ClearAllTag');

                expect(tag.exists()).toEqual(true);
            });
        });

        describe('When content type filter is specified', () => {
            let wrapper;

            beforeAll(() => {
                wrapper = shallow(<FilterTagList {...propsWithContentType} />);
            });

            it("Then is should match snapshot", () => {
                const json = renderer.create(<FilterTagList {...propsWithContentType} />);

                expect(json).toMatchSnapshot();
            });

            it("And it should contain a content type tag", () => {
                const tag = wrapper.find('ContentTypeTag');

                expect(tag.exists()).toEqual(true);
            });

            it('And it should contain a clear all tag', () => {
                const tag = wrapper.find('ClearAllTag');

                expect(tag.exists()).toEqual(true);
            });
        });

        describe("When sub facet filters are specified", () => {
            let wrapper;

            beforeAll(() => {
                wrapper = shallow(<FilterTagList {...propsWithSubFacets} />);
            });

            it("Then it should match snapshot", () => {
                const json = renderer.create(<FilterTagList {...propsWithSubFacets} />);

                expect(json).toMatchSnapshot();
            });

            it("And it should contain a sub facet tag", () => {
                const tag = wrapper.find('SubFacetTags');

                expect(tag.exists()).toEqual(true);
            });

            it("And it should contain a content type tag", () => {
                const tag = wrapper.find('ContentTypeTag');

                expect(tag.exists()).toEqual(true);
            });

            it('And it should contain a clear all tag', () => {
                const tag = wrapper.find('ClearAllTag');

                expect(tag.exists()).toEqual(true);
            });
        });
    });

    describe('Scenario: User Interaction', () => {
        let wrapper;

        beforeAll(() => {
            wrapper = shallow(<FilterTagList {...propsWithAllFilters} />);
        });

        describe("When the clear all tag is clicked", () => {
            it("Then is should call the clear all handler property", () => {
                const tag = wrapper.find('ClearAllTag');

                tag.simulate("remove");

                expect(propsWithAllFilters.filterTagsEvents.onClearAll).toHaveBeenCalled();
            });
        });

        describe("When the keyword tag is clicked", () => {
            it("Then is should call the remove keyword handler property", () => {
                const tag = wrapper.find('KeywordTag');

                tag.simulate("remove");

                expect(propsWithAllFilters.filterTagsEvents.onKeywordRemove).toHaveBeenCalled();
            });
        });

        describe("When the content type tag is clicked", () => {
            it("Then is should call the remove content type handler property", () => {
                const tag = wrapper.find('ContentTypeTag');

                tag.simulate("remove");

                expect(propsWithAllFilters.filterTagsEvents.onContentTypeRemove).toHaveBeenCalled();
            });
        });

        describe("When the sub facet tag is clicked", () => {
            it("Then is should call the remove sub facet handler property", () => {
                const tag = wrapper.find('SubFacetTags');

                tag.props().removeTag();

                expect(propsWithAllFilters.filterTagsEvents.onSubFacetRemove).toHaveBeenCalled();
            });
        });
    });
});