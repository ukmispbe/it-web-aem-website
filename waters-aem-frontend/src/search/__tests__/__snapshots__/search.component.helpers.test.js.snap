// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: Aside Component Scenario: Rendering When children are provided Then it should match snapshot 1`] = `
<div
  className="container__left cmp-search__sort-filter"
>
  <div
    className="cmp-search-hide-btn clearfix"
  >
    <a
      className="btn-hide-sort-filter"
      href="javascript:void(0);"
      onClick={[MockFunction]}
    >
      <div />
      Sort and Filter
    </a>
  </div>
  <div
    className="cmp-search-apply-btn"
  >
    <a
      className="btn-apply-sort-filter"
      href="javascript:void(0);"
      onClick={[Function]}
    >
      Show 0 Results
    </a>
  </div>
  <div
    className="cmp-search-done-btn"
  >
    <a
      className="btn-done-sort-filter"
      href="javascript:void(0);"
      onClick={[Function]}
    >
      Done
    </a>
  </div>
  <div
    className="cmp-search__sort-filter__container"
  >
    <div
      className="cmp-search-sortby"
    >
      <h3>
        Sort By
      </h3>
      <div
        className="css-1pcexqc-container"
        onKeyDown={[Function]}
      >
        <div
          className="css-1xe0he8-control cmp-custom-dropdown__control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1hwfws3 cmp-custom-dropdown__value-container cmp-custom-dropdown__value-container--has-value"
          >
            <div
              className="css-0 cmp-custom-dropdown__single-value"
            >
              Best Match
            </div>
            <input
              className="css-gj7qu5-dummyInput"
              disabled={false}
              id="react-select-3-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={true}
              tabIndex="0"
              value=""
            />
          </div>
          <div
            className="css-1wy0on6 cmp-custom-dropdown__indicators"
          >
            <span
              className="css-1hyfx7x cmp-custom-dropdown__indicator-separator"
            />
            <div
              aria-hidden="true"
              className="css-16pqwjk-indicatorContainer cmp-custom-dropdown__indicator cmp-custom-dropdown__dropdown-indicator"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="child"
    />
  </div>
</div>
`;

exports[`Feature: Aside Component Scenario: Rendering When no children are provided Then it should match snapshot 1`] = `
<div
  className="container__left cmp-search__sort-filter"
>
  <div
    className="cmp-search-hide-btn clearfix"
  >
    <a
      className="btn-hide-sort-filter"
      href="javascript:void(0);"
      onClick={[MockFunction]}
    >
      <div />
      Sort and Filter
    </a>
  </div>
  <div
    className="cmp-search-apply-btn"
  >
    <a
      className="btn-apply-sort-filter"
      href="javascript:void(0);"
      onClick={[Function]}
    >
      Show 0 Results
    </a>
  </div>
  <div
    className="cmp-search-done-btn"
  >
    <a
      className="btn-done-sort-filter"
      href="javascript:void(0);"
      onClick={[Function]}
    >
      Done
    </a>
  </div>
  <div
    className="cmp-search__sort-filter__container"
  >
    <div
      className="cmp-search-sortby"
    >
      <h3>
        Sort By
      </h3>
      <div
        className="css-1pcexqc-container"
        onKeyDown={[Function]}
      >
        <div
          className="css-1xe0he8-control cmp-custom-dropdown__control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="css-1hwfws3 cmp-custom-dropdown__value-container cmp-custom-dropdown__value-container--has-value"
          >
            <div
              className="css-0 cmp-custom-dropdown__single-value"
            >
              Best Match
            </div>
            <input
              className="css-gj7qu5-dummyInput"
              disabled={false}
              id="react-select-2-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              readOnly={true}
              tabIndex="0"
              value=""
            />
          </div>
          <div
            className="css-1wy0on6 cmp-custom-dropdown__indicators"
          >
            <span
              className="css-1hyfx7x cmp-custom-dropdown__indicator-separator"
            />
            <div
              aria-hidden="true"
              className="css-16pqwjk-indicatorContainer cmp-custom-dropdown__indicator cmp-custom-dropdown__dropdown-indicator"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Feature: FilterTagList Component Scenario: Rendering When content type filter is specified Then is should match snapshot 1`] = `
<div
  className="cmp-search-filters__tags clearfix"
>
  <a
    className="cmp-search-filters__tags__clear"
    href="javascript:void(0);"
    onClick={[Function]}
  >
    <div />
    <span>
      Clear All
    </span>
  </a>
  <a
    href="javascript:void(0);"
    onClick={[Function]}
  >
    <div />
    <span>
      Content Type: Application Note
    </span>
  </a>
</div>
`;

exports[`Feature: FilterTagList Component Scenario: Rendering When keyword filter is specified Then it should match spapshot 1`] = `
<div
  className="cmp-search-filters__tags clearfix"
>
  <a
    className="cmp-search-filters__tags__clear"
    href="javascript:void(0);"
    onClick={[Function]}
  >
    <div />
    <span>
      Clear All
    </span>
  </a>
  <a
    href="javascript:void(0);"
    onClick={[Function]}
  >
    <div />
    <span>
      Keyword: hplc
    </span>
  </a>
</div>
`;

exports[`Feature: FilterTagList Component Scenario: Rendering When keyword spelling suggestion is specified Then it should match spapshot 1`] = `
<div
  className="cmp-search-filters__tags clearfix"
>
  <a
    className="cmp-search-filters__tags__clear"
    href="javascript:void(0);"
    onClick={[Function]}
  >
    <div />
    <span>
      Clear All
    </span>
  </a>
  <a
    href="javascript:void(0);"
    onClick={[Function]}
  >
    <div />
    <span>
      Keyword: HPLC
    </span>
  </a>
</div>
`;

exports[`Feature: FilterTagList Component Scenario: Rendering When no filters are specified Then it should match snapshot 1`] = `
<div
  className="cmp-search-filters__emptytags"
/>
`;

exports[`Feature: FilterTagList Component Scenario: Rendering When sub facet filters are specified Then it should match snapshot 1`] = `
<div
  className="cmp-search-filters__tags clearfix"
>
  <a
    className="cmp-search-filters__tags__clear"
    href="javascript:void(0);"
    onClick={[Function]}
  >
    <div />
    <span>
      Clear All
    </span>
  </a>
  <a
    href="javascript:void(0);"
    onClick={[Function]}
  >
    <div />
    <span>
      Content Type: Application Note
    </span>
  </a>
</div>
`;

exports[`Feature: Menu Component Scenario: Rendering When showing facet menu Then it should match snapshot 1`] = `
<div
  className="facet-menu-container"
>
  <div
    className="facet-menu-container__heading"
  >
    <div
      className="cmp-search-filters__emptytags"
    />
    <div
      className="heading--with-selected-value"
    >
      <div
        className="back-btn"
      >
        <a
          href="javascript:void(0)"
          onClick={[MockFunction]}
        >
          <div />
           
          filter By
        </a>
      </div>
      <h3>
        
      </h3>
    </div>
  </div>
  <div
    className="facet-menu-container__body"
  >
    <div
      className="cmp-search-filters"
      id="js-search-filters"
    >
      <ul />
    </div>
  </div>
</div>
`;

exports[`Feature: Menu Component Scenario: Rendering When showing the content type menu Then it should match snapshot 1`] = `
<div
  className="content-type-menu-container"
>
  <div
    className="content-type-menu-container__heading"
  >
    filter By
  </div>
  <div
    className="content-type-menu-container__body"
  />
</div>
`;
